feature-libs/product-configurator/rulebased/root/interceptor/cpq-access-storage.service.ts

```diff
@@ -0,0 +1,132 @@
+import { Injectable } from '@angular/core';
+import { AuthService } from '@spartacus/core';
+import { BehaviorSubject, Observable, Subscription, timer } from 'rxjs';
+import {
+  distinctUntilChanged,
+  expand,
+  filter,
+  switchMap,
+  take,
+} from 'rxjs/operators';
+import { CpqAccessData } from './cpq-access-data.models';
+import { CpqAccessLoaderService } from './cpq-access-loader.service';
+import { CpqConfiguratorAuthConfig } from './cpq-configurator-auth.config';
+
+@Injectable({ providedIn: 'root' })
+export class CpqAccessStorageService {
+  protected readonly EXPIRED_TOKEN = {
+    accessToken: 'INVALID DUMMY',
+    accessTokenExpirationTime: 0,
+  };
+
+  constructor(
+    protected cpqAccessLoaderService: CpqAccessLoaderService,
+    protected authService: AuthService,
+    protected config: CpqConfiguratorAuthConfig
+  ) {}
+
+  protected cpqAccessObservable: Observable<CpqAccessData>;
+  protected currentCpqAccessSubscription: Subscription;
+  protected cpqAccessDataCache: BehaviorSubject<CpqAccessData>;
```

Comment:
For consistency let's suffix all observables (including behavior subjects) with `$`. 
Btw. what about renaming them to?:
`cpqAccessDataCache` -> `_cpqAccessData$`
`cpqAccessObservable` -> `cpqAccessData$`

Reply:
pick